File: docs/07-archive/legacy-repos/ebs_bo/src/bo/db/models/output_preset.py
from sqlmodel import SQLModel, Field

from .base import utcnow


class OutputPreset(SQLModel, table=True):
    __tablename__ = "output_presets"
    preset_id: int | None = Field(default=None, primary_key=True)
    name: str = Field(nullable=False, unique=True)
    output_type: str = Field(default="ndi")
    width: int = Field(default=1920)
    height: int = Field(default=1080)
    framerate: int = Field(default=60)
    security_delay_sec: int = Field(default=0)
    chroma_key: bool = Field(default=False)
    is_default: bool = Field(default=False)
    created_at: str = Field(default_factory=utcnow)
    updated_at: str = Field(default_factory=utcnow)

File: team1-frontend/lib/models/entities/output_preset.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'output_preset.freezed.dart';
part 'output_preset.g.dart';

@freezed
class OutputPreset with _$OutputPreset {
  const factory OutputPreset({
    @JsonKey(name: 'presetId') required int presetId,
    required String name,
    @JsonKey(name: 'outputType') required String outputType,
    required int width,
    required int height,
    required int framerate,
    @JsonKey(name: 'securityDelaySec') required int securityDelaySec,
    @JsonKey(name: 'chromaKey') required bool chromaKey,
    @JsonKey(name: 'isDefault') required bool isDefault,
  }) = _OutputPreset;

  factory OutputPreset.fromJson(Map<String, dynamic> json) =>
      _$OutputPresetFromJson(json);
}
